### PR TITLE
MON-63843 engine accept connections from centreon monitoring agent

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

REFS:MON-63843

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: engine/modules/opentelemetry/doc/opentelemetry.md [hunk 1]
```diff
@@ -207,3 +207,97 @@ An example of configuration:
...
+Event if all protobuf objects are opentelemetry objects, grpc communication is made in streaming mode. It is more efficient, it allows reverse connection (engine can connect to an agent running in a DMZ) and 
+Engine can send configuration on each config update.
+You can find all grpc definitions are agent/proto/agent.proto.
+Every time engine configuration is updated, we calculate configuration for each connected agent and send it on the wire if we find a difference with the old configuration. That's why each connection has a ```agent::MessageToAgent _last_config``` attribute.
+So, the opentelemetry engine server supports two services, opentelemetry service and agent streaming service.
+OpenTelemetry data is different from telegraf one:
+* host service attributes are stored in resource_metrics.resource.attributes
+* performance data (min, max, critical lt, warning gt...) is stored in exemplar, service status is stored in status metric
+  
+Example:
```
Comment (bouda1) [inline engine/modules/opentelemetry/doc/opentelemetry.md:221]: Could you also give the service output just as given by the plugin, so it is easier to compare.

File: engine/modules/opentelemetry/doc/opentelemetry.md [hunk 2]
```diff
@@ -207,3 +207,97 @@ An example of configuration:
...
+  ```protobuf
+    uint32 check_interval = 2;
+    //limit the number of active checks in order to limit charge
+    uint32 max_concurrent_checks = 3;
+    //period of metric exports (in seconds)
+    uint32 export_period = 4;
+    //after this timeout, process is killed (in seconds)
+    uint32 check_timeout = 5;
+  ```
+* A list of services that agent have to check
```
Comment (bouda1) [inline engine/modules/opentelemetry/doc/opentelemetry.md:299]:
```suggestion
* A list of services that agent has to check
```

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_check_result_builder.hh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                    {
+                        "asDouble": 100,
+                        "filteredAttributes": [
+                            {
+                                "key": "max"
+                            }
+                        ]
+                    }
+                ],
+                "asInt": "30"
```
Comment (bouda1) [inline engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_check_result_builder.hh:85]: Just to be sure of my understanding, it should be:
```
"asInt": 30
```

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_config.hh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * Centreon Engine is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Centreon Engine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
```
Comment (bouda1) [inline engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_config.hh:18]: wrong license

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_impl.hh [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+ * as published by the Free Software Foundation.
+ *
+ * Centreon Engine is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Centreon Engine. If not, see
+ * <http://www.gnu.org/licenses/>.
```
Comment (bouda1) [inline engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_impl.hh:17]: wrong license

File: engine/modules/opentelemetry/doc/opentelemetry.md (hunk 1 above)
Comment (jean-christophe81) [reply engine/modules/opentelemetry/doc/opentelemetry.md:221]: done

File: engine/modules/opentelemetry/doc/opentelemetry.md [hunk 6]
```diff
@@ -207,3 +207,97 @@ An example of configuration:
         }
     }
   ```
+
+### centreon monitoring agent
+Event if all protobuf objects are opentelemetry objects, grpc communication is made in streaming mode. It is more efficient, it allows reverse connection (engine can connect to an agent running in a DMZ) and
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/doc/opentelemetry.md:212]: don't see difference

File: engine/modules/opentelemetry/doc/opentelemetry.md (hunk 2 above)
Comment (jean-christophe81) [reply engine/modules/opentelemetry/doc/opentelemetry.md:299]: done

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_check_result_builder.hh (hunk 3 above)
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_check_result_builder.hh:85]: fixed

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_impl.hh (hunk 5 above)
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/centreon_agent/agent_impl.hh:17]: done

File: engine/modules/opentelemetry/doc/opentelemetry.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/doc/opentelemetry.md:212]: done

**Review by bouda1 — APPROVED** [review]

**Pull request merged**